### ISSUE TITLE
Complex numbers implement

### ISSUE THREAD

**Issue by 6112562a** [SciSharp/Numpy.NET#51]
This is an awesome project!
I'm working on both Python and CSharp.
Recently,I'm trying to translate a python project to CSharp, however I need to implement complex numbers.
Hope Numpy.NET can handle Complex numbers.
Thanks.

**Comment by henon** [comment]
not yet, but we can make it happen ;)

**Comment by henon** [comment]
I just got array creation np.array(...) to work with complex numbers:

```c#

        [TestMethod]
        public void ComplexNumbers()
        {
            //>>> a = np.array([1+2j, 3+4j, 5+6j])
            //>>> a.imag
            //array([2.,  4.,  6.])
            var a = np.array(new Complex[] { new Complex(1, 2), new Complex(3,4), new Complex(5,6), });
            Assert.AreEqual("array([1., 3., 5.])", a.real.repr);
            Assert.AreEqual("array([2., 4., 6.])", a.imag.repr);
            //>>> np.imag(a)
            //array([2., 4., 6.])
            //>>> np.real(a)
            //array([1., 3., 5.])
            Assert.AreEqual("array([1., 3., 5.])", np.real(a).repr);
            Assert.AreEqual("array([2., 4., 6.])", np.imag(a).repr);
            //>>> a.imag = np.array([8, 10, 12])
            //>>> a
            //array([1. +8.j,  3.+10.j,  5.+12.j])
            a.imag = np.array(new []{8, 10, 12});
            Assert.AreEqual("array([1. +8.j, 3.+10.j, 5.+12.j])", a.repr);
            //>>> np.imag(1 + 1j)
            //1.0
            Assert.AreEqual(1.0, np.imag(new Complex(1, 1)).asscalar<double>());
        }
```

what else do you need? I am guessing you need .GetData<Complex>() etc

**Comment by henon** [comment]
This also works now: `a.GetData<Complex>();` and returns a Complex[]

**Comment by henon** [comment]
released as numpy v1.22 on nuget. please try it out. if you need further support for complex numbers, let me know.

**Comment by 6112562a** [comment]
Yeah,Greet work!Thanks a lot.
